connect_cmd.go

```diff
@@ -127,77 +156,102 @@ func connectAction(ctx *cli.Context) error {
 				err)}
 		if uiSettings.isMachineReadable {
 			connectResult.RHSMConnectError = errorMessages["rhsm"].message.Error()
+			connectResult.ContentEnabled = false
 		} else {
 			fmt.Printf(
-				"%v Cannot connect to Red Hat Subscription Management\n",
+				" [%v] Cannot connect to Red Hat Subscription Management\n",
+				uiSettings.iconError,
+			)
+			fmt.Printf(
+				"  [%v] Skipping generation of Red Hat repository file\n",
 				uiSettings.iconError,
 			)
 		}
 	} else {
 		connectResult.RHSMConnected = true
-		interactivePrintf("%v %v\n", uiSettings.iconOK, returnedMsg)
+		interactivePrintf(" [%v] %v\n", uiSettings.iconOK, returnedMsg)
+		if ContentFeature.Enabled {
+			if uiSettings.isMachineReadable {
+				connectResult.ContentEnabled = true
+			}
+			interactivePrintf("  [%v] Content ... Red Hat repository file generated\n", uiSettings.iconOK)
+		} else {
+			if uiSettings.isMachineReadable {
+				connectResult.ContentEnabled = false
+			}
+			interactivePrintf("  [ ] Content ... Red Hat repository file not generated\n")
+		}
 	}
 	durations["rhsm"] = time.Since(start)
 
 	/* 2. Register insights-client */
-	if errors, exist := errorMessages["rhsm"]; exist {
-		if errors.level == log.LevelError {
-			interactivePrintf(
-				"%v Skipping connection to Red Hat Insights\n",
-				uiSettings.iconError,
-			)
-		}
-	} else {
-		start = time.Now()
-		err = showProgress(" Connecting to Red Hat Insights...", registerInsights)
-		if err != nil {
-			connectResult.InsightsConnected = false
-			errorMessages["insights"] = LogMessage{
-				level:   log.LevelError,
-				message: fmt.Errorf("cannot connect to Red Hat Insights: %w", err)}
-			if uiSettings.isMachineReadable {
-				connectResult.InsightsError = errorMessages["insights"].message.Error()
-			} else {
-				fmt.Printf("%v Cannot connect to Red Hat Insights\n", uiSettings.iconError)
+	if AnalyticsFeature.Enabled {
+		if errors, exist := errorMessages["rhsm"]; exist {
+			if errors.level == log.LevelError {
+				interactivePrintf(
+					"  [%v] Skipping connection to Red Hat Insights\n",
+					uiSettings.iconError,
+				)
 			}
 		} else {
-			connectResult.InsightsConnected = true
-			interactivePrintf("%v Connected to Red Hat Insights\n", uiSettings.iconOK)
+			start = time.Now()
+			err = showProgress(" Connecting to Red Hat Insights...", registerInsights)
```

Comment:
With the introduction of the space-indented results printout in the interactive
mode, the in-progress spinner looks awkward.

```
 [✓] Connected to Red Hat Subscription Management
  [✓] Content ... Red Hat repository file generated
 / Connecting to Red Hat Insights...
```

Can we put that spinner inside brackets too? That can reduce the bouncing back
and forth as new lines are printed.

```
 [✓] Connected to Red Hat Subscription Management
  [✓] Content ... Red Hat repository file generated
  [/] Connecting to Red Hat Insights...
```

Reply:
Yes, it is doable.